scripts/sct_maths.py

```diff
@@ -472,7 +506,69 @@ def laplacian(data, sigmas):
     # from scipy.ndimage.filters import laplace
     # return laplace(data.astype(float))
 
+def compute_similarity(data1, data2, fname_out='', metric='', verbose=1):
+    assert data1.size == data2.size, "\n\nERROR: the data don't have the same size.\nPlease use  \"sct_register_multimodal -i im1.nii.gz -d im2.nii.gz -identity 1\"  to put the input images in the same space"
+    data1_1d = data1.ravel()
+    data2_1d = data2.ravel()
+
+    if metric == 'mi':
+        res = mutual_information(data1_1d, data2_1d, normalized=True)
+        metric_full = 'Mutual information'
+    if metric == 'corr':
+        res = correlation(data1_1d, data2_1d)
+        metric_full = 'Pearson correlation coefficient'
+
+    printv('\n'+ metric_full +': ' + str(res), verbose, 'info')
+
+    path_out, filename_out, ext_out = extract_fname(fname_out)
+    if ext_out not in ['.txt', '.pkl', '.pklz', '.pickle']:
+        printv('ERROR: the output file should a text file or a pickle file. Received extension: '+ext_out, 1, 'error')
+
+    elif ext_out == '.txt':
+        file_out = open(fname_out, 'w')
+        file_out.write(metric_full+': \n'+str(res))
+        file_out.close()
+
+    else:
+        import pickle, gzip
+        if ext_out == '.pklz':
+            pickle.dump(res, gzip.open(fname_out, 'wb'), protocol=2)
+        else:
+            pickle.dump(res, open(fname_out, 'w'), protocol=2)
+
+def mutual_information(x, y, nbins=32, normalized=False):
```

Comment:
The function documentation could also use details on what is returned and what is `nbins`.
`
"""Compute the mutual information between clusters x and y.

:param x: numpy.array cluster
:param y: numpy.array cluster
:param nbins: number of bins for not normalized mutual information
:param normalized: flag if x and y are normalized 
:returns: float non-negative value 
`

If you feel that the function parameters are straight forward, it is ok to delete the param from the documentation.